mlir/python/mlir/dialects/tensor.py

```diff
@@ -43,6 +44,18 @@ def __init__(
         super().__init__(result_type, dynamic_sizes, loc=loc, ip=ip)
 
 
+def empty(
+    sizes: Sequence[Union[int, Value]],
+    element_type: Type,
+    *,
+    loc=None,
+    ip=None,
+) -> _ods_cext.ir.Value:
+    return _get_op_result_or_op_results(
+        EmptyOp(sizes=sizes, element_type=element_type, loc=loc, ip=ip)
+    )
```

Comment:
Doesn't this already exist? I.e. this should already be generated (with basically this exact same signature):

```
# tensor.py

@_ods_cext.register_operation(_Dialect)
class EmptyOp(_ods_ir.OpView):
  OPERATION_NAME = "tensor.empty"

  _ODS_REGIONS = (0, True)

  def __init__(self, result, dynamicSizes, *, loc=None, ip=None):
    operands = []
    results = []
    attributes = {}
    regions = None
    operands.extend(_get_op_results_or_values(dynamicSizes))
    _ods_context = _ods_get_default_loc_context(loc)
    results.append(result)
    _ods_successors = None
    super().__init__(self.build_generic(attributes=attributes, results=results, operands=operands, successors=_ods_successors, regions=regions, loc=loc, ip=ip))

def empty(result, dynamic_sizes, *, loc=None, ip=None) -> _ods_ir.Value:
  return _get_op_result_or_op_results(EmptyOp(result=result, dynamicSizes=dynamic_sizes, loc=loc, ip=ip))
```

Reply:
hi, @makslevental the signature is different, I know the existence of the generated `tensor.empty`, but we do not have any usage of that API, instead, what we are using is the extended `class EmptyOp(EmptyOp)` not the class `EmptyOp(_ods_ir.OpView)` , the signature is a little different and it was easier to use. so I am thinking adding `tensor.empty` method for this extended class.
After this, if users want to use the `tensor.empty`, they can change from  `tensor.empty(RankedTensorType.get([3, 4], f32), [])` to ` tensor.empty([3, 4], f32)`
```
@_ods_cext.register_operation(_Dialect, replace=True)
class EmptyOp(EmptyOp):
    """Extends the tensor.empty op."""
    def __init__(
        self,
        sizes: Sequence[Union[int, Value]],
        element_type: Type,
        *,
        loc=None,
        ip=None,
    ):
```
I found a code similar to this for extended `arith.ConstantOp` which has a manual `arith.constant` https://github.com/llvm/llvm-project/blob/16e22a6d49ee148b426cb4285aa479b327f3b0f7/mlir/python/mlir/dialects/arith.py#L107-L110